### PR TITLE
AB#38280 highlight active query

### PR DESCRIPTION
**Pull request by gracet9182**
## Overview

* Couldn't previously see which query was selected
* Now highlights the previously selected query
* ADO board link: [Task #38280](https://dev.azure.com/microsoftgarage/Intern%20GitHub/_sprints/taskboard/GI21%20-%20Graph%20Explorer/Intern%20GitHub/Y21-S/04?workitem=38280)

## Demo
* It is now highlighted 

https://user-images.githubusercontent.com/47228306/123651287-31052900-d7e0-11eb-99af-3cfb87f54853.mp4

### CODE REVIEW

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 1]
```diff
@@ -197,6 +207,7 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
         default:
           return (
             <TooltipHost
+
```
Comment (lhao03) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: is this randomly added extra line? can it be removed?

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 2]
```diff
@@ -220,19 +231,28 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
...
+    if (this.state.selectedQuery === props.item.id) {
+      customStyles.root = { backgroundColor: '#C0C0C0' };
+    }
 
     if (props) {
       if (!tokenPresent && props.item.method !== 'GET') {
         selectionDisabled = true;
       }
       return (
+
```
Comment (lhao03) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: random extra line? can it be removed?

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 3]
```diff
@@ -220,19 +231,28 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
...
         <div className={classes.groupHeader}>
           <DetailsRow
+
```
Comment (lhao03) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: random extra line? can this be removed (as per the others)?

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 4]
```diff
@@ -59,10 +65,14 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
     } else {
       this.props.actions!.fetchSamples();
     }
+    const { sampleQueries } = this.state;
+    this.groups = generateGroupsFromList(sampleQueries, 'category');
```
Comment (lhao03) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: why was this added? when i remove these lines it still works

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 5]
```diff
@@ -123,7 +133,7 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
                 tooltipProps={{
                   onRenderContent: () => (
                     <div style={{ paddingBottom: 3 }}>
-                      <FormattedMessage id={signInText} />
+                      <FormattedMessage id={"signInText"} />
```
Comment (lhao03) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: why were the quotes added. i think its referring to the variable `signInText` above.
image

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 6]
```diff
@@ -220,6 +230,10 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
     const { tokenPresent } = this.props;
     const classes = classNames(this.props);
     let selectionDisabled = false;
+    const customStyles: Partial<IDetailsRowStyles> = {};
+    if (this.state.selectedQuery === props.item.id) {
+      customStyles.root = { backgroundColor: '#C0C0C0' };
```
Comment (lhao03) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: can this be put in a css or scss file?

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 7]
```diff
@@ -448,7 +467,7 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
           items={sampleQueries}
           selectionMode={SelectionMode.none}
           columns={columns}
-          groups={groups}
+          groups={this.groups}
```
Comment (lhao03) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: why was `this` added?

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx (hunk 4 above)
Comment (gracet9182) [reply src/app/views/sidebar/sample-queries/SampleQueries.tsx]: It seems to regenerate the groups every render so adding this prevented the group from defaulting the collapse every time someone selected a query.

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx (hunk 5 above)
Comment (gracet9182) [reply src/app/views/sidebar/sample-queries/SampleQueries.tsx]: Nice catch!!

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx (hunk 7 above)
Comment (gracet9182) [reply src/app/views/sidebar/sample-queries/SampleQueries.tsx]: I saved it as a variable instead cause it was regenerating the groups

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx (hunk 6 above)
Comment (gracet9182) [reply src/app/views/sidebar/sample-queries/SampleQueries.tsx]: Thanks for pointing this out! I might have to check with the GE team because the other styling in this file is also in this format, not sure if I should be following their convention. Is it okay to leave it like this for now until then?

Comment (lhao03) [reply src/app/views/sidebar/sample-queries/SampleQueries.tsx]: yeah if theres other css in the file then its fine!

**Review by lhao03 — APPROVED** [review]
lgtm! i accidently requested myself to review this again 
YikesMonkeyLookAndLeaveGIF

**Review by omersayshi — APPROVED** [review]
good work!

**Review by ayubun — COMMENTED** [review]
Tested and works on my local ヾ(≧▽≦*)o

One comment I want your opinion on;

Do you think that typing in the input box should remove this CSS? Technically speaking, changing the query at all would result in it no longer matching that highlighted sample query, so maybe a simple unhighlight for typing behavior in the input box could be a good addition.

**Comment by gracet9182** [comment]
> Tested and works on my local ヾ(≧▽≦*)o
> 
> One comment I want your opinion on;
> 
> Do you think that typing in the input box should remove this CSS? Technically speaking, changing the query at all would result in it no longer matching that highlighted sample query, so maybe a simple unhighlight for typing behavior in the input box could be a good addition.

Thanks for bringing it up! I was thinking about it before too and this feature is supposed encompass more of the "previously selected" query. As in, even if I changed something in the input box, I might still want to go back to where I last was when I selected via UI (it's hard to trace this back via the history tab). Do you think it's valid?

**Comment by anneguo3** [comment]
> > Tested and works on my local ヾ(≧▽≦*)o
> > One comment I want your opinion on;
> > Do you think that typing in the input box should remove this CSS? Technically speaking, changing the query at all would result in it no longer matching that highlighted sample query, so maybe a simple unhighlight for typing behavior in the input box could be a good addition.
> 
> Thanks for bringing it up! I was thinking about it before too and this feature is supposed encompass more of the "previously selected" query. As in, even if I changed something in the input box, I might still want to go back to where I last was when I selected via UI (it's hard to trace this back via the history tab). Do you think it's valid?

Intent behind this feature is to highlight last selected query - I think picking up on changes in the textbox is a nuance that isn't worth it because some requests require you to add in an ID (which could be picked up as a change) yet you would still "technically" be on said sample query - so I think existing behaviour is fine IMO

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx [hunk 8]
```diff
@@ -45,10 +46,15 @@ import { sidebarStyles } from '../Sidebar.styles';
 import { isJsonString } from './sample-query-utils';
 
 export class SampleQueries extends Component<ISampleQueriesProps, any> {
+  resetCollapse: boolean;
```
Comment (csyan5) [inline src/app/views/sidebar/sample-queries/SampleQueries.tsx]: better to have explicit modifier specified for every class member such as `private _resetCollapse: boolean;` also it's a convention to add `_` for private members

**Review by csyan5 — APPROVED** [review]

File: src/app/views/sidebar/sample-queries/SampleQueries.tsx (hunk 8 above)
Comment (gracet9182) [reply src/app/views/sidebar/sample-queries/SampleQueries.tsx]: Thanks!!

**Comment by ayubun** [comment]
> Thanks for bringing it up! I was thinking about it before too and this feature is supposed encompass more of the "previously selected" query. As in, even if I changed something in the input box, I might still want to go back to where I last was when I selected via UI (it's hard to trace this back via the history tab). Do you think it's valid?

Omg, I don't know why I missed seeing this! Almost feels like I have too many GH notis...

I know what you mean. I am actually not sure which one is the better option. Because it is something that I am not sure of (and I think it is opinionated, to be honest) I think both are fine and it is better to leave it as is c:

Good job on the PR again btws! I know I am commenting after merging 😅
